blockly-ws-search/workspace_search.js

```diff
@@ -114,44 +128,90 @@ class WorkspaceSearch {
     // Create the button
     var btn = document.createElement('button');
     Blockly.utils.dom.addClass(btn, className);
-    Blockly.bindEventWithChecks_(btn,
-        'click', this, onClickFn);
+    Blockly.bindEventWithChecks_(btn, 'click', this, onClickFn);
     btn.append(textSpan);
     return btn;
   }
 
+  /**
+   * Handles input value change in search bar.
+   * @param {Event} e The oninput event.
+   */
+  onInput_(e) {
+    if (this.searchOnInput) {
+      const inputValue = e.target.value;
+      if (inputValue !== this.searchText_) {
+        this.setSearchText_(inputValue);
+        this.search();
+      }
+    }
+  }
+
   /**
    * Handles a key down for the search bar.
-   * @param {Event} e The key down event.
+   * @param {KeyboardEvent} e The key down event.
    * @private
    */
   onKeyDown_(e) {
-    if (e.keyCode == Blockly.utils.KeyCodes.ESC) {
-      console.log("Close search bar");
-    } else if (e.keyCode == Blockly.utils.KeyCodes.TAB) {
-      return;
-    } else {
-      // Should check that the text value has changed before running search.
-      console.log("Search all the things");
+    if (e.key === 'Escape') {
+      this.close();
+    } else if (e.key === 'Enter') {
+      if (this.searchOnInput) {
+        this.next_();
+      } else {
+        this.setSearchText_(e.target.value);
+        this.search();
+      }
     }
   }
 
   /**
-   * Goes to the previous block.
-   * @param {Event} e The key down event.
+   * Sets search text.
+   * @param {string} text
    * @private
    */
-  previous_(e) {
+  setSearchText_(text) {
+    this.searchText_ = text.trim()
+  }
+
+  /**
+   * Selects the previous block.`
+   * @private
+   */
+  previous_() {
+    if (!this.blocks_.length) {
+      return;
+    }
+    this.selectBlock_(this.blockIndex_ - 1);
     console.log("Get previous value");
   }
 
   /**
-   * Goes to the next block.
-   * @param {Event} e The key down event.
+   * Selects the next block.
    * @private
    */
   next_() {
-    console.log("Get next values");
+    if (!this.blocks_.length) {
+      return;
+    }
+    this.selectBlock_(this.blockIndex_ + 1);
+    console.log("Get next value");
+  }
+
+  /**
+   * Selects the block at the given index.
+   * @param {number} index Index of block to select. Number is wrapped.
+   */
+  selectBlock_(index) {
```

Comment:
Not necessary for this release, but I think we should either pass in the list of blocks here, or we should make this.blocks_ public. That way people can easily override if they want to change behavior.

Reply:
Good point, we should probably revisit a lot of the methods. I forgot subclasses aren't supposed to access private variables.